airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -200,7 +228,16 @@ def _should_retry(self, response: requests.Response) -> bool:
         """
         if self.error_handler is None:
             return response.status_code == 429 or 500 <= response.status_code < 600
-        return bool(self.interpret_response_status(response).action == ResponseAction.RETRY)
+
+        if self.use_cache:
+            interpret_response_status = self.interpret_response_status
+        else:
+            # Use a tiny cache to limit the memory footprint. It doesn't have to be large because we mostly
+            # only care about the status of the last response received
+            # Cache the result because the HttpStream first checks if we should retry before looking at the backoff time
+            interpret_response_status = lru_cache(maxsize=10)(self.interpret_response_status)
```

Comment:
why was the lru_cache moved from an annotation to here? I assume it's because we don't want to use the lru cache with the request cache. can you comment on why that would be a problem?

Reply:
We cannot use both `request_cache` and `lru_cache` at the same time, because in this case, with `use_cache=True`, two response classes will be passed to the `interpret_response_status` method: [OriginalResponse](https://github.com/requests-cache/requests-cache/blob/main/requests_cache/models/response.py#L50) and [CachedResponse](https://github.com/requests-cache/requests-cache/blob/main/requests_cache/models/response.py#L70), and both of them do not support the `__hash__` method, so we will catch an error for responses that either should be cached or that we got from a cache file.